src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts

```diff
@@ -761,10 +761,25 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 		}
 	}
 
-	private setRecentlyUsedTask(task: Task): void {
-		const key = task.getRecentlyUsedKey();
+	private async setRecentlyUsedTask(task: Task): Promise<void> {
+		let key = task.getRecentlyUsedKey();
 		if (!InMemoryTask.is(task) && key) {
-			this.getRecentlyUsedTasks().set(key, JSON.stringify(this.createCustomizableTask(task)));
+			const customizations = this.createCustomizableTask(task);
+			if (ContributedTask.is(task) && customizations) {
+				let custom: CustomTask[] = [];
+				let customized: IStringDictionary<ConfiguringTask> = Object.create(null);
+				await this.computeTasksForSingleConfig(task._source.workspaceFolder ?? this.workspaceFolders[0], {
+					version: '2.0.0',
+					tasks: [customizations]
+				}, TaskRunSource.System, custom, customized, TaskConfig.TaskConfigSource.TasksJson, true);
+				for (const configuration in customized) {
+					// There should only be one configuration in customized, but check that it matches the original task just to be sure.
```

Comment:
What happens now if we do have a programming error somewhere and the keys are not the same. The we use the key from the beginning which to my understanding would be wrong. So shouldn't we simply assert this or ignore the task if something goes wrong.

Reply:
Yes, it is effectively ignored later when the recents are read. However it would be better to ignore earlier (here) instead.

Reply:
OK. That makes sense now to me. Are you planning to do any changes or do we go with the ignore later approach?

Reply:
I made the change for ignoring here since it is better not to defer the problem.